client/pages/newTask.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import parseRoute from '../lib/parse-route';
+
+export default class NewTask extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      taskName: '',
+      statusId: '',
+      dateCreated: '',
+      deadline: '',
+      userId: '',
+      description: '',
+      projectId: ''
+    };
+    this.handleChangeTaskName = this.handleChangeTaskName.bind(this);
+    this.handleChangeDateCreated = this.handleChangeDateCreated.bind(this);
+    this.handleChangeDeadline = this.handleChangeDeadline.bind(this);
+    this.handleChangeDescription = this.handleChangeDescription.bind(this);
+    this.handleChangeStatus = this.handleChangeStatus.bind(this);
+    this.handleChangeUserId = this.handleChangeUserId.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    const route = parseRoute(window.location.hash);
+    this.setState({ projectId: Number(route.params.get('projectId')) });
+  }
+
+  handleChangeTaskName(event) {
+    this.setState({ taskName: event.target.value });
+  }
+
+  handleChangeStatus(event) {
+    this.setState({ statusId: Number(event.target.value) });
+  }
+
+  handleChangeDateCreated(event) {
+    this.setState({ dateCreated: event.target.value });
+  }
+
+  handleChangeDeadline(event) {
+    this.setState({ deadline: event.target.value });
+  }
+
+  handleChangeUserId(event) {
+    this.setState({ userId: Number(event.target.value) });
+  }
+
+  handleChangeDescription(event) {
+    this.setState({ description: event.target.value });
+  }
```

Comment:
If you give each of your inputs a `name` prop, you can create one change handler

```js
handleChange(event) {
  const { name, value } = event.target
  const newState = {}
  newState[name] = value
  this.setState(newState)
}
```